### PR TITLE
Another small collection of fixes for problems raised by clang.

### PR DESCRIPTION
**Pull request by tworaz**
This is yet another small collection of fixes for various problems observed when building the code with LLVM based toolchain. The problems are usually pretty generic and should not have any negative effects on GCC builds.

Most of the fixes collected here have been burried in the depths of my main Genode work branch, so they are pretty old and have been tested for quite a while.

### CODE REVIEW

File: repos/os/src/server/event_filter/button_scroll_source.h [hunk 1]
```diff
@@ -182,8 +182,8 @@ class Event_filter::Button_scroll_source : public Source, Source::Filter
 					 * execute both conditions regardless of the result of the
 					 * first call of 'handle_activation'.
 					 */
-					if (_vertical_wheel  .handle_deactivation(event)
-					  | _horizontal_wheel.handle_deactivation(event)) {
+					if (_vertical_wheel  .handle_deactivation(event) ||
+					    _horizontal_wheel.handle_deactivation(event)) {
```
Comment (nfeske) [inline repos/os/src/server/event_filter/button_scroll_source.h:186]: The bitwise OR is indeed intended here. Please consider the comment above the `if`.

**Comment by nfeske** [comment]
Thank you @tworaz for this wonderfully documented commit series. I merged all commits except for the bitwise-or change (I left a comment at the commit).

Note that I slightly abbreviated some of the first lines of the commit messages to make them fit into 50 characters. Do you approve?

**Comment by tworaz** [comment]
@nfeske The commits you've merged to staging look fine to me.

Sorry for not reading the comment in button_scroll_source.h. This bitwise or warning is only raised by the upcoming clang 14 release. I'll see if I can find some solution that will satisfy clang and not break the code in the process. This also happens the only place in the code this warning is raised for to I can as well just ignore it.

**Comment by nfeske** [comment]
We could introduce two bool constants along those lines (just a sketch):
```
bool const vertical_deactivated   = _vertical_wheel  .handle_deactivation(event);
bool const horizontal_deactivated = _horizontal_wheel.handle_deactivation(event);
if (vertical_deactivated | horizontal_deactivated)
  ...
```

It is a bit more code but becomes probably easier to follow and alleviates the need for the comment above.

**Comment by tworaz** [comment]
This would probably work, but I'd like to first update my clang 14 snapshot to the recently branched off llvm-14 version. The one I use now is over a month old and new warnings do get relaxed a bit closer to release. For all intents and purposes lets treat this PR as done. I'll create a new one if the final clang/llvm 14 branch still causes problems.

**Comment by nfeske** [comment]
Alright. The commits just entered the master branch.

**Pull request closed without merging**